File: Programa1/Python/problema1.py
"""
Solución del Problema 1.

Dado un arreglo de personas y una bolsa de vaquitas,
se determina quiénes alcanzan vaquitas, considerando
solo a las personas a las que sí les gustan.
"""

from modelos import Persona, BolsaVaquitas


def problema1(personas, bolsa):
    """
    Resuelve el problema 1.

    Recorre la lista de personas y entrega vaquitas
    mientras haya disponibles. Imprime los nombres
    de las personas que sí alcanzaron.

    Complejidad: O(n)
    """
    print("Personas que recibieron vaquitas:")
    for persona in personas:
        if persona.le_gustan_vaquitas:
            if bolsa.dar_vaquita():
                print(persona.nombre)
            else:
                break

File: Programa1/Java/Problema1.java
/**
 * Solución del Problema 1.
 * Se reparten vaquitas solo a las personas a las que les gustan,
 * mientras haya disponibles.
 *
 * @author Claudia Yarinet Osorio Lopez
 */
public class Problema1 {

    /**
     * Recorre el arreglo de personas y entrega vaquitas.
     * Imprime los nombres de quienes sí alcanzaron.
     */
    public static void resolver(Persona[] personas, BolsaVaquitas bolsa) {
        System.out.println("Personas que recibieron vaquitas:");

        for (Persona p : personas) {
            if (p.leGustanVaquitas()) {
                if (bolsa.darVaquita()) {
                    System.out.println(p.getNombre());
                } else {
                    break;
                }
            }
        }
    }
}
